apis/ecr/v1alpha1/repository_types.go

```diff
@@ -0,0 +1,122 @@
+)
+
+// RepositoryParameters define the desired state of an AWS Elastic Container Repository
+type RepositoryParameters struct {
+
+	// The image scanning configuration for the repository. This determines whether
+	// images are scanned for known vulnerabilities after being pushed to the repository.
+	// +optional
+	ImageScanningConfiguration *ImageScanningConfiguration `json:"imageScanningConfiguration,omitempty"`
+
+	// The tag mutability setting for the repository. If this parameter is omitted,
+	// the default setting of MUTABLE will be used which will allow image tags to
+	// be overwritten. If IMMUTABLE is specified, all image tags within the repository
+	// will be immutable which will prevent them from being overwritten.
+	// +optional
+	// +kubebuilder:validation:Enum=MUTABLE;IMMUTABLE
+	ImageTagMutability *string `json:"imageTagMutability,omitempty"`
+
+	// Metadata tagging key value pairs
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
+}
+
+// A RepositorySpec defines the desired state of a Elastic Container Repository.
+type RepositorySpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	// +optional
+	ForProvider RepositoryParameters `json:"forProvider"`
+}
+
+// RepositoryObservation keeps the state for the external resource
+type RepositoryObservation struct {
+	// The date and time, in JavaScript date format, when the repository was created.
+	CreatedAt *metav1.Time `json:"createdAt,omitempty"`
+
+	// The image scanning configuration for a repository.
+	ImageScanningConfiguration ImageScanningConfiguration `json:"imageScanningConfiguration,omitempty"`
+
+	// The tag mutability setting for the repository.
+	ImageTagMutability string `json:"imageTagMutability,omitempty"`
+
+	// The AWS account ID associated with the registry that contains the repository.
+	RegistryID string `json:"registryId"`
+
+	// The Amazon Resource Name (ARN) that identifies the repository. The ARN contains
+	// the arn:aws:ecr namespace, followed by the region of the repository, AWS
+	// account ID of the repository owner, repository namespace, and repository
+	// name. For example, arn:aws:ecr:region:012345678910:repository/test.
+	RepositoryArn string `json:"repositoryArn,omitempty"`
+
+	// The name of the repository.
+	RepositoryName string `json:"repositoryName,omitempty"`
+
+	// The URI for the repository. You can use this URI for container image push
+	// and pull operations.
+	RepositoryURI string `json:"repositoryUri,omitempty"`
+}
+
+// ImageScanningConfiguration Scanning Configuration
+type ImageScanningConfiguration struct {
+
+	// The setting that determines whether images are scanned after being pushed
+	// to a repository. If set to true, images will be scanned after being pushed.
+	// If this parameter is not specified, it will default to false and images will
+	// not be scanned unless a scan is manually started with the StartImageScan
+	// API.
+	ScanOnPush bool `json:"scanOnPush,omitempty"`
```

Comment:
I will typically make a single field in an embedded struct required (i.e. no `omitempty` here) if its parent is optional so that supplying the parent necessitates supplying the single child.